contracts/peg/PegStabilityModuleNonCustodial.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {IPCVDeposit} from "./../pcv/IPCVDeposit.sol";
+import {FixedPricePSM} from "./FixedPricePSM.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @notice Peg Stability Module that holds no funds.
+/// On a mint, it transfers all proceeds to a PCV Deposit and then deposits them into the deposit's target
+/// When funds are needed for a redemption, they are simply pulled from the PCV Deposit
+contract PegStabilityModuleNonCustodial is FixedPricePSM {
+    using SafeERC20 for IERC20;
+
+    /// @notice constructor
+    /// @param _floor minimum acceptable orace price in basis points
+    /// @param _ceiling maximum acceptable orace price in basis points
+    /// @param params PSM constructor parameter struct
+    /// @param _mintFeeBasisPoints fee for minting in basis points
+    /// @param _redeemFeeBasisPoints fee for redemption in basis points
+    /// @param _feiLimitPerSecond the amount of FEI that will be replenished every second on the buffer
+    /// @param _mintingBufferCap cap on minting buffer
+    /// @param _underlyingToken underlying token this PSM trades against
+    /// @param _pcvDeposit where all assets are stored and all assets are pulled from
+    constructor(
+        uint256 _floor,
+        uint256 _ceiling,
+        OracleParams memory params,
+        uint256 _mintFeeBasisPoints,
+        uint256 _redeemFeeBasisPoints,
+        uint256 _feiLimitPerSecond,
+        uint256 _mintingBufferCap,
+        IERC20 _underlyingToken,
+        IPCVDeposit _pcvDeposit
+    ) FixedPricePSM(
+        _floor,
+        _ceiling,
+        params,
+        _mintFeeBasisPoints,
+        _redeemFeeBasisPoints,
+        0, /// hardcode reserve threshold to 0
+        _feiLimitPerSecond,
+        _mintingBufferCap,
+        _underlyingToken,
+        _pcvDeposit
```

Comment:
Are we going to have one psm per pcv deposit? This doesn't make much sense to me.

Reply:
I don't believe this design requires that we have one PSM per PCV deposit. 

Each `NonCustodial` PSM will deposit it's funds into a designated initial PCV Deposit. However, we can later freely withdraw those funds and transfer them to a different PCVDeposit using the `ratioPCVController`.